plugins/modules/iosxr_logging_global.py

```diff
@@ -0,0 +1,403 @@
+
+"""
+The module file for iosxr_logging_global
+"""
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: iosxr_logging_global
+version_added: 2.4.0
+short_description: Manages logging attributes of Cisco IOSXR network devices
+description: This module manages the logging attributes of Cisco IOSXR network devices
+notes:
+- Tested against IOSXR 7.0.2.
+- This module works with connection C(network_cli).
+author: Ashwini Mhatre (@amhatre)
+options:
+  config:
+    description: A dictionary of logging options.
+    type: dict
+    suboptions:
+      archive:
+        description: logging to a persistent device(disk/harddisk)
+        type: dict
+        suboptions:
+          device:
+            type: str
+            description: Configure the archive device
+          archive_length:
+            type: int
+            description: The maximum no of weeks of log to maintain.
+          archive_size:
+            type: int
+            description: The total size of the archive.
+          file_size:
+            type: int
+            description: The maximum file size for a single log file..
+          frequency:
+            type: str
+            description: The collection interval for logs.
+            choices: ["daily", "weekly"]
+          severity: &severity
+            description: Logging severity level
+            type: str
+            choices:
+              - alerts
+              - critical
+              - debugging
+              - emergencies
+              - errors
+              - informational
+              - notifications
+              - warnings
+          threshold:
+            type: int
+            description: Threshold percent <1-99>.
+      buffered:
+        description: Set buffered logging parameters
+        type: dict
+        suboptions:
+          size: &size
+            description: Logging buffer size
+            type: int
+          severity: *severity
+          discriminator: &discriminator
+            description: Establish MD-Buffer association
+            type: list
+            elements: dict
+            suboptions:
+              match_params:
+                type: str
+                description: Set match/no-match discriminator.
+                choices: ["match1", "match2", "match3", "nomatch1", "nomatch2", "nomatch3"]
+              name:
+                type: str
+                description: discriminator name.
+      console:
+        description: Set console logging parameters
+        type: dict
+        suboptions:
+          severity: &severity1
+            description: Logging severity level
+            type: str
+            choices:
+              - alerts
+              - critical
+              - debugging
+              - emergencies
+              - errors
+              - informational
+              - notifications
+              - warning
+              - disable
```

Comment:
AFAIU, `disable` is not a severity. It turns on/off console logging. We can have a subkey called `state` with `choices: ["enabled", "disabled"]` to cover this.

Reply:
ack